docs: add informational note for cloudflare_managed_headers

jramosf opened the pull request:
This change makes it easier for the user on how to find the Managed Transforms that can be used (it will include a blue [callout](https://developer.hashicorp.com/terraform/registry/providers/docs#callouts))

jacobbednarz commented:
thanks for the PR! unfortunately, the file you've edited is a generated artifact of the documentation process. to see how to make changes to the documentation, you're best to check out https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/registry-documentation.md

jramosf commented:
> thanks for the PR! unfortunately, the file you've edited is a generated artifact of the documentation process. to see how to make changes to the documentation, you're best to check out https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/registry-documentation.md

Thanks for hinting that out, I've rebased the PR and moved the documentation to the appropriate golang file `internal/sdkv2provider/resource_cloudflare_managed_headers.go`. I've confirmed the documentation in Markdown is generated with `make docs` as well.
<img width="1398" alt="image" src="https://github.com/user-attachments/assets/aa31d622-2075-4c8a-a503-7fd158395fae">

jramosf commented:
Hey @jacobbednarz is something else required here? Thanks!

jacobbednarz commented:
thanks but i'm not sure what value this is adding so i'm hesitant to merge it. the link above yours in the existing description is already going to developer docs that has this link as well as a bunch more context.

jramosf commented:
@jacobbednarz I'm closing the PR then if this is not going to be merged. Feel free to reopen if otherwise. Thanks,
